newton/tests/test_model.py

```
@@ -486,12 +492,16 @@ def test_collision_filter_pairs_canonical_order(self):
 
         # Verify all collision filter pairs are in canonical order (s1 < s2)
         for s1, s2 in model.shape_collision_filter_pairs:
-            self.assertLess(s1, s2, f"Collision filter pair ({s1}, {s2}) is not in canonical order")
+            self.assertLess(
+                mode.shape_type[s1],
```

Ruff (0.15.11)

[error] Undefined name `mode`

(F821)